### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by vladgusakovskiy**
[DEMO LINK](https://vladgusakovskiy.github.io/react_keyboard/)

### CODE REVIEW

**Review by roma-shafihullin — APPROVED** [review]
good solution 👍

**Review by denys-danyliuk — CHANGES_REQUESTED** [review]
GJ
let's use event.code to check if the pressed key is space and write [spac
Знімок екрана 2022-09-12 о 13 18 30
e] if so

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,39 @@
...
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.handleDocumentEvent);
+  }
+
+  handleDocumentEvent = (event: KeyboardEvent) => {
+    if (event.key === 'space') {
+      this.setState({ pressedKey: 'The last pressed key is [space]' });
+    } else {
+      this.setState({ pressedKey: `The last pressed key is [${event.key}]` });
```
Comment (denys-danyliuk) [inline src/App.tsx:24]: don't save a whole string to pressedKey, save only a name of the key

**Review by denys-danyliuk — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,43 @@
...
+
+  handleDocumentEvent = (event: KeyboardEvent) => {
+    this.setState({ pressedKey: event.key });
+
+    if (this.state.pressedKey === ' ') {
+      this.state.pressedKey = 'space';
+    }
+  };
+
+  render(): React.ReactNode {
```
Comment (denys-danyliuk) [inline src/App.tsx:28]:
```suggestion
  render() {
```

**Review by denys-danyliuk — APPROVED** [review]
Good job, you probably forgot to do deploy, because when I press space I see empty space

**Pull request closed without merging**